source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -26,21 +23,20 @@ public class MessageExchangeStream : IMessageExchangeStream
         readonly ILog log;
         readonly StreamWriter streamWriter;
         readonly StreamReader streamReader;
-        readonly JsonSerializer serializer;
+        readonly IMessageSerializer serializer;
         readonly Version currentVersion = new Version(1, 0);
 
-        public MessageExchangeStream(Stream stream, IEnumerable<Type> registeredServiceTypes, ILog log)
+        public MessageExchangeStream(Stream stream, IMessageSerializer serializer, ILog log)
```

Comment:
I really like how much cleaner this class is with the IMessageSerializer 👍  SOLID for the win! 🥳